packages/admin/admin-rte/src/core/extension/NonBreakingSpace/ToolbarButton.tsx

```diff
@@ -0,0 +1,37 @@
+import { NonBreakingSpace } from "@comet/admin-icons";
+import Tooltip from "@mui/material/Tooltip";
+import { EditorState, Modifier } from "draft-js";
+import * as React from "react";
+import { FormattedMessage } from "react-intl";
+
+import ControlButton from "../../Controls/ControlButton";
+import { IControlProps } from "../../types";
+
+const NO_BREAK_SPACE_UNICODE_CHAR = 0x00a0;
+
+function ToolbarButton({ editorState, setEditorState }: IControlProps): React.ReactElement {
+    function handleClick(event: React.MouseEvent) {
+        event.preventDefault(); // Preserve focus in editor
+
+        const currentContent = editorState.getCurrentContent();
+        const selection = editorState.getSelection();
+        // TODO insert \u00a0 in a way that link-entities dont break when inserted in the middle of a link text
```

Comment:
is this something that can be fixed later (aka never)?

Reply:
I straight up copied this from the soft hyphen character and replaced the char code 😅 

Tested three scenarios and looked at the editor state:

1. Insert character with toolbar button: One entity, two entity ranges
2. Insert character using keyboard (on Mac: Option+Space): One entity, two entity ranges
3. Insert character using copy/paste: One entity, one entity range. However, the SVG icon is not shown at all, it's just a normal nbsp

I doubt we'll ever fix this, especially since we're planing on ditching Draft.js one day. And even if we decide to fix it, I believe it has to do something with this TODO: https://github.com/vivid-planet/comet/blob/main/packages/admin/cms-admin/src/blocks/createRichTextBlock.tsx#L29.

So I suggest to remove this TODO. What do you think?

Reply:
in this case, leave the todo imho